### ISSUE TITLE
Install with existing stores defined in config.php

### ISSUE THREAD

**Issue by FreekVandeursen** [magento/magento2#15686]
### Preconditions
1. Core Magento, no additional modules
2. Tested on 2.2-develop with PHP 7.0 and 2.3-develop with PHP 7.1

### Steps to reproduce
1. Create app/etc/config.php, with additional store, e.g.
```<?php
return [
    'modules' => [
       /* All core modules enabled */
      ...
    ],
    'scopes' => [
        'websites' => [
            'admin' => [
                'website_id' => '0',
                'code' => 'admin',
                'name' => 'Admin',
                'sort_order' => '0',
                'default_group_id' => '0',
                'is_default' => '0'
            ],
            'base' => [
                'website_id' => '1',
                'code' => 'base',
                'name' => 'Main Website',
                'sort_order' => '0',
                'default_group_id' => '1',
                'is_default' => '1'
            ]
        ],
        'groups' => [
            [
                'group_id' => '0',
                'website_id' => '0',
                'code' => 'default',
                'name' => 'Default',
                'root_category_id' => '0',
                'default_store_id' => '0'
            ],
            [
                'group_id' => '1',
                'website_id' => '1',
                'code' => 'main_website_store',
                'name' => 'Main Website Store',
                'root_category_id' => '2',
                'default_store_id' => '1'
            ]
        ],
        'stores' => [
            'admin' => [
                'store_id' => '0',
                'code' => 'admin',
                'website_id' => '0',
                'group_id' => '0',
                'name' => 'Admin',
                'sort_order' => '0',
                'is_active' => '1'
            ],
            'default' => [
                'store_id' => '1',
                'code' => 'default',
                'website_id' => '1',
                'group_id' => '1',
                'name' => 'Default Store View',
                'sort_order' => '0',
                'is_active' => '1'
            ],
            'test' => [
                'store_id' => '3',
                'code' => 'test',
                'website_id' => '1',
                'group_id' => '1',
                'name' => 'Test',
                'sort_order' => '1',
                'is_active' => '1'
            ]
        ]
    ]
];
```

2. run bin/magento setup:install

### Expected result
1. Magento being installed, and additional store created

### Actual result
The result depends on the store_id chosen for the additional store(s)
With store_id 2:
```
[SUCCESS]: Magento installation complete.
[SUCCESS]: Magento Admin URI: /admin_2zf3dj
Processing configurations data from configuration file...
Stores were processed
```

With any store_id > 2:
```
[SUCCESS]: Magento installation complete.
[SUCCESS]: Magento Admin URI: /admin_w6717j
Processing configurations data from configuration file...
Import failed: Requested store is not found
```

The same problem occurs when creating additional websites or store groups, with non-sequential id's

**Comment by jalogut** [comment]
Hi @FreekVandeursen 

I faced the same issue an another one if the default store has a name different than "default".

I created a PR. Maybe it helps you

- https://github.com/magento/magento2/pull/16317

**Comment by engcom-backlog-nickolas** [comment]
Hello @FreekVandeursen, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by magento-engcom-team** [comment]
Hi @FreekVandeursen. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-24597` was closed as `non-reproducible` in `2.2-develop`.
It means that Magento team either unable to reproduce this issue using provided _Steps to Reproduce_ from the **_Description section_** on clean Magento instance or the issue has been already fixed in the scope of other tasks.

But if you still run into this problem please update or provide additional information/steps/preconditions in the **_Description section_** and reopen this issue.
